algorithms/src/Kokkos_Random.hpp

```diff
@@ -1626,7 +1626,7 @@ template <class ExecutionSpace, class ViewType, class RandomPool,
 void fill_random(const ExecutionSpace& exec, ViewType a, RandomPool g,
                  typename ViewType::const_value_type begin,
                  typename ViewType::const_value_type end) {
-  Impl::apply_to_view_of_static_rank(
+  Impl::ApplyToViewOfStaticRank<ViewType>::apply(
```

Comment:
```suggestion
  // Using Impl::apply_to_view_of_static_rank function here
  // is not compatible with modules. Using the partial specialization works.
  Impl::ApplyToViewOfStaticRank<ViewType>::apply(
```

I would like a comment something like the suggestion for posterior.